### PR TITLE
Clean up and optimize images

### PR DESCRIPTION
**Pull request by smarterclayton**
* openvswitch, by inheriting from node, saves an extra set of layers
* node, by not chmoding scripts, saves 300M
* fail the build if an image fails
* force base image to be built in release.sh
* remove old haproxy-base image

### CODE REVIEW

**Comment by smarterclayton** [comment]
@tdawson @sdodson @stevekuznetsov @danwinship openshift node was 300M larger than it needed to be, see the chown commits as well as the openvswitch switch.

**Comment by smarterclayton** [comment]
[test]

File: images/openvswitch/Dockerfile [hunk 1]
```diff
@@ -3,16 +3,12 @@
...
 
 COPY scripts/* /usr/local/bin/
 
-RUN curl -L -o /etc/yum.repos.d/origin-next-epel-7.repo https://copr.fedoraproject.org/coprs/maxamillion/origin-next/repo/epel-7/maxamillion-origin-next-epel-7.repo && \
-    INSTALL_PKGS="openvswitch" && \
-    yum install -y $INSTALL_PKGS && \
-    rpm -V $INSTALL_PKGS && \
-    yum clean all && \
-    chmod +x /usr/local/bin/*
+RUN INSTALL_PKGS="openvswitch" && \
```
Comment (stevekuznetsov) [inline images/openvswitch/Dockerfile]: I understand the intent but does this make a layer and isn't it useless?

Comment (sdodson) [inline images/openvswitch/Dockerfile]: openvswitch is already in the node image, just drop this.

Comment (sdodson) [inline images/openvswitch/Dockerfile]: Also, +1 to switching the base of openvswitch to be the node. If you need openvswitch image you need the node image so may as well just make it the same with different entrypoint.

**Comment by smarterclayton** [comment]
I wanted to keep the check in case someone altered the node image accidentally.

File: images/openvswitch/Dockerfile (hunk 1 above)
Comment (tdawson) [inline images/openvswitch/Dockerfile]: You might have already noticed this, but I'm just going to put it in as a comment so others will.
Yes, openvswitch is already in node, or it should be.  This little blip isn't installing openvswitch, it is verifying that it's there.
So this is a nice check incase we manage to grab a node image that doesn't have openvswitch for some reason.

**Comment by sdodson** [comment]
Ok, Fine with me.

**Comment by smarterclayton** [comment]
[test] with new image builder

**Comment by smarterclayton** [comment]
[merge] based on passing tests

**Comment by smarterclayton** [comment]
Backporting to 1.5 release branch

**Comment by smarterclayton** [comment]
Also, we need a good way to prevent image size regressions - we can't do it
in build-images since we trying not to need the docker binary, and the test
suites are too late.  Might be an extra task we run from Jenkins

On Mar 11, 2017, at 11:40 AM, OpenShift Bot <notifications@github.com>
wrote:

Merged #13335 <https://github.com/openshift/origin/pull/13335>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/13335#event-996021551>, or mute
the thread

**Pull request merged**